### PR TITLE
Fix aria-label documentation, removing statement that says it should be used only for interactive items

### LINKED ISSUE #28397: aria-label can be used on non-interactive elements
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label

### What specific section or headline is this issue about?

_No response_

### What information was incorrect, unhelpful, or incomplete?

>The aria-label attribute defines a string value that labels an interactive element.

>The aria-label attribute is intended for interactive elements only.

> **Note:** `aria-label` is intended for use on interactive elements, or elements made to be interactive via other ARIA declarations, when there is no appropriate text visible in the DOM that could be referenced as a label

### What did you expect to see?

For the first sentence, I would expect the mdn page to use the same definition as the [Accessible Rich Internet Applications (WAI-ARIA) 1.2 W3C Recommendation](https://www.w3.org/TR/wai-aria-1.2/#aria-label).

>Defines a string value that labels the current element.

For the other paragraphs, I would simply delete them since it's not accurate (and contradicts the rest of the doc).

### Do you have any supporting links, references, or citations?

Official spec: https://www.w3.org/TR/wai-aria-1.2/#aria-label

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/accessibility/aria/attributes/aria-label`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/accessibility/aria/attributes/aria-label/index.md
* Last commit: https://github.com/mdn/content/commit/07f0cf4375aaa02e1071d8bd0e8518db7609b7a9
* Document last modified: 2023-07-17T23:05:13.000Z

</details>

### PR DESCRIPTION
**Pull request by phsantiago**
### Description
This change closes https://github.com/mdn/content/issues/28397, by changing the first paragraph of aria-label page, based on aria-label spec https://w3c.github.io/aria/#aria-label.

### Motivation
There are noninteractive elements, such as List, Img, and Meter, that allow aria-label, and it's useful to have this context announced when using assistive technologies.
- https://w3c.github.io/aria/#list aria-label is present in "Inherited States and Properties"
- https://w3c.github.io/aria/#img aria-label is present in "Inherited States and Properties"
- https://w3c.github.io/aria/#meter aria-label is present in "Inherited States and Properties"

### Additional details
The confusion might have appeared because there are multiple non interactive aria roles, which aria-label is forbidden:

> Used in Roles: All elements of the base markup except for the following roles: [caption](https://w3c.github.io/aria/#caption), [code](https://w3c.github.io/aria/#code), [definition](https://w3c.github.io/aria/#definition), [deletion](https://w3c.github.io/aria/#deletion), [emphasis](https://w3c.github.io/aria/#emphasis), [generic](https://w3c.github.io/aria/#generic), [insertion](https://w3c.github.io/aria/#insertion), [mark](https://w3c.github.io/aria/#mark), [none](https://w3c.github.io/aria/#none), [paragraph](https://w3c.github.io/aria/#paragraph), [strong](https://w3c.github.io/aria/#strong), [subscript](https://w3c.github.io/aria/#subscript), [suggestion](https://w3c.github.io/aria/#suggestion), [superscript](https://w3c.github.io/aria/#superscript), [term](https://w3c.github.io/aria/#term), [time](https://w3c.github.io/aria/#time)

### Related issues and pull requests
https://github.com/mdn/content/issues/28397

### CODE REVIEW

**Comment by scottaohara** [comment]
Might make sense to explicitly mention aria-label can be used to name an element, so long as the elements implicit or explicit aria role doesn’t prohibit naming.  

Listitem, for example being a role that is incorrectly not in the name prohibited category, but will be once the spec updates to correct this. But the other example, role=list or main, article, etc are all nameable roles/elements.

File: files/en-us/web/accessibility/aria/attributes/aria-label/index.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ spec-urls: https://w3c.github.io/aria/#aria-label
 
 {{AccessibilitySidebar}}
 
-The `aria-label` attribute defines a string value that labels an interactive element.
+The `aria-label` attribute defines a string value that names an element.
```
Comment (phsantiago) [inline files/en-us/web/accessibility/aria/attributes/aria-label/index.md]: > Might it make sense to explicitly mention that aria-label can be used to name an element, so long as the element's implicit or explicit aria role doesn’t prohibit naming.

Pushed a change https://github.com/mdn/content/pull/35465/commits/83297c784da87276c534ff5019dcca5f761f6de3 making it clearer that label should only be used with roles that don't prohibit author naming.

>Listitem, for example being a role that is incorrectly not in the name prohibited category, but will be once the spec updates to correct this. But the other example, role=list or main, article, etc are all nameable roles/elements.

Thanks for the context, @scottaohara! I found this nice PR of you on the theme: https://github.com/w3c/aria/pull/2292. Let me know if I could help with anything there, too   🙂

File: files/en-us/web/accessibility/aria/attributes/aria-label/index.md [hunk 2]
```diff
@@ -7,13 +7,13 @@ spec-urls: https://w3c.github.io/aria/#aria-label
 
 {{AccessibilitySidebar}}
 
-The `aria-label` attribute defines a string value that labels an interactive element.
+The `aria-label` attribute defines a string value that can be used to name an element, as long as the element's role does not [prohibit naming](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label#associated_roles).
```
Comment (phsantiago) [inline files/en-us/web/accessibility/aria/attributes/aria-label/index.md:10]: > Might it make sense to explicitly mention that aria-label can be used to name an element, so long as the element's implicit or explicit aria role doesn’t prohibit naming.

Pushed a change https://github.com/mdn/content/pull/35465/commits/83297c784da87276c534ff5019dcca5f761f6de3 making it clearer that label should only be used with roles that don't prohibit author naming.

>Listitem, for example being a role that is incorrectly not in the name prohibited category, but will be once the spec updates to correct this. But the other example, role=list or main, article, etc are all nameable roles/elements.

Thanks for the context, @scottaohara! I found this nice PR of you on the theme: https://github.com/w3c/aria/pull/2292. Let me know if I could help with anything there, too   🙂

**Comment by phsantiago** [comment]
@scottaohara I’ve addressed the feedback provided in the initial review of PR. Could you please take some time to re-review? Thanks in advance 🙂

File: files/en-us/web/accessibility/aria/attributes/aria-label/index.md [hunk 3]
```diff
@@ -32,7 +32,7 @@ The code below shows an example of how to use the `aria-label` attribute to prov
 </button>
 ```
 
-> **Note:** `aria-label` is intended for interactive elements or for elements made interactive via other ARIA declarations when there's no visible text in the DOM to serve as a label.
+> **Note:** `aria-label` is intended for naming elements that allow `aria-label` or for elements made interactive via other ARIA declarations when there's no visible text in the DOM to serve as a label.
```
Comment (scottaohara) [inline files/en-us/web/accessibility/aria/attributes/aria-label/index.md:35]:
```suggestion
> **Note:** `aria-label` is intended for naming elements where the implicit or explicit role does not prohibit naming. It is strongly recommended to prioritize the use of `aria-labelledby` over `aria-label` if a visible label exists for the element to reference and receive its name from. 
```

**Comment by phsantiago** [comment]
Hi @scottaohara, thanks for your previous review. I applied the change you suggested https://github.com/mdn/content/pull/35465/commits/0ead6ef8737421e4323f290ef1d82952dec270a3

Would you mind taking another look when you have a moment? I really appreciate your time and effort in helping get this PR merged

**Comment by phsantiago** [comment]
Hi @scottaohara, just following up on this PR as it’s been a few weeks since my last re-review request. I’ve addressed the feedback, and would really appreciate it if you could take another look when you have time. Thanks again!

**Comment by Josh-Cena** [comment]
Perhaps @estelle could also review this, if @scottaohara could not

**Review by scottaohara — APPROVED** [review]

**Pull request merged**